### ISSUE TITLE
ROG Phone 3 comparability

### ISSUE THREAD

**Issue by jeanrivera** [programminghoch10/MobileIconChanger#1]
It appears not compatible with my device stock A11 (R) Rom, I wonder if there is any kind of log I can provide you to help you find why it isn't compatible and if possible fix on a future release.

Obviously I have tried reboot after install, and tried disabling other modules that mess with SystemUI to make sure it wasn't a conflict issue.

![Screenshot_20210701-002817847](https://user-images.githubusercontent.com/24832354/124064667-3e4e2d80-da03-11eb-8000-78a015221260.jpg)

**Comment by programminghoch10** [comment]
Did you enable the resource hooks?
Are you using LSPosed or EdXposed?

**Comment by jeanrivera** [comment]
> Did you enable the resource hooks?
> Are you using LSPosed or EdXposed?

Yes and LSPosed

**Comment by jeanrivera** [comment]
some new development on the issue, now it does show me the icons to replace, but it does not replace them after systemUI reboot or after normal reboot
![Screenshot_20210701-135046083](https://user-images.githubusercontent.com/24832354/124169181-ec91bb80-da73-11eb-8c83-e1e094eaa52c.jpg)

[logcat.txt](https://github.com/programminghoch10/MobileIconChanger/files/6749930/logcat.txt)

**Comment by programminghoch10** [comment]
Please install version `2.2.1` and capture a logcat again.
Your system _seems to be_ compatible but the replacement code fails somehow.
Version `2.2.1` has more debugging output and should give us more intel.
If all does not work, it is probably because your system uses custom icons (compare it to the AOSP example screenshots; original android does not have activity indicators).

**Comment by jeanrivera** [comment]
> Please install version `2.2.1` and capture a logcat again.
> Your system _seems to be_ compatible but the replacement code fails somehow.
> Version `2.2.1` has more debugging output and should give us more intel.
> If all does not work, it is probably because your system uses custom icons (compare it to the AOSP example screenshots; original android does not have activity indicators).

I guess it will not work for me :-( , I have sent my systemUI on the first post, but not sure if that will help at all

[logcat.txt](https://github.com/programminghoch10/MobileIconChanger/files/6751224/logcat.txt)

**Comment by jeanrivera** [comment]
Do you want me to close this issue here and keep troubleshooting on XDA?

**Comment by programminghoch10** [comment]
No, I prefer it here. I had not yet the chance to look into the latest logcat, please give me some time.

**Comment by programminghoch10** [comment]
Okay I looked at the logcat and I don't understand what happend there...
The module loses track of the replacement icon at a point where that should not be possible...
Anyways I've attached a test apk with even more logging output. (rename it to .apk first, github only let me upload zip files)
[MobileIconChanger-ROGPhone3test1.apk.zip](https://github.com/programminghoch10/MobileIconChanger/files/6758429/MobileIconChanger-ROGPhone3test1.apk.zip)
Please also select _one_ icon to be replaced by another system icon (select any icon from category `system`).
Then as always, restart systemui while capturing a logcat.

Oh also I've already had a look at your `SystemUI.apk`, it all looks to be compatible.

**Comment by jeanrivera** [comment]
thanks for your time and for trying to debug this issue, here is the new logcat, still not working for me 
[logcat.txt](https://github.com/programminghoch10/MobileIconChanger/files/6759045/logcat.txt)

**Comment by programminghoch10** [comment]
This is very weird.
The code runs normally and replaces everything as it should do.
The execution time is a bit wanky, but I don't think that is a problem.
It will be very hard to debug this issue further without the phone at hand.

You can try one more thing, as I said in the last comment

> Please also select one icon to be replaced by another system icon (select any icon from category system).

If that works, maybe there is still a chance.

**Comment by jeanrivera** [comment]
> This is very weird.
> The code runs normally and replaces everything as it should do.
> The execution time is a bit wanky, but I don't think that is a problem.
> It will be very hard to debug this issue further without the phone at hand.
> 
> You can try one more thing, as I said in the last comment
> 
> > Please also select one icon to be replaced by another system icon (select any icon from category system).
> 
> If that works, maybe there is still a chance.

The last log I provided is replacing icons with system icons already, I think we have hit a wall with this issue. If there is nothing else we can do about it, let me know if you want me to close this issue

**Comment by programminghoch10** [comment]
You did replace some with system icons, but the code for replacing them did not run.
I think that the replacement code only is run when the icon is actually accessed by the system.
Can you try replacing an icon where you can actually observe if it changes?

Yes you are right it looks like we've hit a wall. 
But you do know I can also close the issue myself? 
Or you can just close it when you think it's not worth your time examining further.
Either way there is no need to hurry.

**Comment by jeanrivera** [comment]
> You did replace some with system icons, but the code for replacing them did not run.
> I think that the replacement code only is run when the icon is actually accessed by the system.
> Can you try replacing an icon where you can actually observe if it changes?
> 
> Yes you are right it looks like we've hit a wall.
> But you do know I can also close the issue myself?
> Or you can just close it when you think it's not worth your time examining further.
> Either way there is no need to hurry.

I am not in a hurry and willing to provide any logs you need or other system files as request, and also test more for you when needed, if you come up with another idea or require any other logs please let me know.

**Comment by programminghoch10** [comment]
> **Can you try replacing an icon where you can actually observe if it changes?**

and upload another logcat and this file: `/data/misc/lspd_sB5BapCTQS5o5h0W/prefs/com.programminghoch10.mobileiconchanger/mobileIconChanger.xml`

**Comment by jeanrivera** [comment]
[mobileIconChanger.xml.zip](https://github.com/programminghoch10/MobileIconChanger/files/6835253/mobileIconChanger.xml.zip)

It's fake .zip not an actual archive just delete the .zip at the end
